internal/winapi/process.go

```diff
@@ -1,10 +1,83 @@
 package winapi
 
+import (
+	"errors"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
 const PROCESS_ALL_ACCESS uint32 = 2097151
 
-// DWORD GetProcessImageFileNameW(
-//	HANDLE hProcess,
-//	LPWSTR lpImageFileName,
-//	DWORD  nSize
+const (
+	PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE = 0x20016
+	PROC_THREAD_ATTRIBUTE_JOB_LIST      = 0x2000D
+)
+
+type ProcThreadAttributeList struct {
+	_ [1]byte
+}
+
+// typedef struct _STARTUPINFOEXW {
+// 	STARTUPINFOW                 StartupInfo;
+// 	LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList;
+// } STARTUPINFOEXW, *LPSTARTUPINFOEXW;
+type StartupInfoEx struct {
+	// This is a recreation of the same binding from the stdlib. The x/sys/windows variant for whatever reason
+	// doesn't work when updating the list for the pseudo console attribute. It has the process immediately exit
+	// with exit code 0xc0000142 shortly after start.
+	//
+	// TODO (dcantah): Swap to the x/sys/windows definitions after https://go-review.googlesource.com/c/sys/+/371276/1/windows/exec_windows.go#153
+	// gets in.
+	windows.StartupInfo
+	ProcThreadAttributeList *ProcThreadAttributeList
+}
+
+// NewProcThreadAttributeList allocates a new ProcThreadAttributeList, with
+// the requested maximum number of attributes. This must be cleaned up by calling
+// DeleteProcThreadAttributeList.
+func NewProcThreadAttributeList(maxAttrCount uint32) (*ProcThreadAttributeList, error) {
+	var size uintptr
+	err := initializeProcThreadAttributeList(nil, maxAttrCount, 0, &size)
```

Comment:
is this documented, that it'll give you size if queried with `*procThreadAttributeList` as `nil`? maybe worth a comment

Reply:
Oop missed this. Yes it's listed here although translate NULL to nil for us obviously https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-initializeprocthreadattributelist#remarks

Reply:
I'll add a comment on this PR when integrating this work into it https://github.com/microsoft/hcsshim/pull/1233